### DISCUSSION TITLE
Bcc is not blind

### DISCUSSION THREAD

**Discussion by Rike-cz** [stalwartlabs/mail-builder#27] [Q&A]
Hello, I tried to send e-mails through 3 different SMTP servers and in all cases all recipients can see whole list of _Bcc_. How to manage to have it hidden as usual?
When using different mail building tools and the same servers, there is no problem. Maybe I did something wrong, but using only simple sending as in examples. Can you give me a hint?:-)

**Comment by mdecimus** [comment]
Hi, do not include the addresses in the Bcc field. Only provide the addresses you need to blind copy during the SMTP transaction.

**Reply by Rike-cz** [reply]
Thank you for your reply. I tried to use `SmtpClient::rcpt_to`, but if it's before `SmtpClient::send`, I get only error "_Sender not yet given_". I cannot figure out how to deal with it.

**Reply by Rike-cz** [reply]
@mdecimus Even AI did not find solution:-) I read RFC 2822/5322 and they write: 
>  The "Bcc:" field (where the "Bcc" means "Blind Carbon Copy") contains addresses of recipients of the message whose addresses are not to be revealed to other recipients of the message.  There are three ways in which the "Bcc:" field is used.  In the first case, when a message containing a "Bcc:" field is prepared to be sent, _the "Bcc:" line is removed_ even though all of the recipients (including those specified in the "Bcc:" field) are sent a copy of the message.  In the second case, recipients specified in the "To:" and "Cc:" lines each are sent a copy of the message with _the "Bcc:" line removed as above_, but the recipients on the "Bcc:" line get a separate copy of the message containing a "Bcc:" line. ... Finally, since a "Bcc:" field may contain no addresses, a "Bcc:" field can be sent without any addresses indicating to the recipients that blind copies were sent to someone.  Which method to use with "Bcc:" fields is implementation dependent, but refer to the "Security Considerations" section of this document for a discussion of each.

I think it's SmtpClient task to deal with Bcc corectly, isn't it?

**Reply by link2xt** [reply]
"Sender not yet given" likely means you did not send "MAIL FROM:" yet. You first need to send "MAIL FROM:", then "RCPT TO:".

**Reply by Rike-cz** [reply]
@link2xt Sure, but I cannot "hook" into sending process any other way.
